test/lib/ansible_test/_internal/ansible_util.py

```diff
@@ -214,35 +211,27 @@ def get_ansible_python_path(args):  # type: (CommonConfig) -> str
         os.symlink(ANSIBLE_LIB_ROOT, os.path.join(python_path, 'ansible'))
 
     if not args.explain:
-        generate_egg_info(python_path)
+        generate_dist_info()
 
     get_ansible_python_path.python_path = python_path  # type: ignore[attr-defined]
 
     return python_path
 
 
-def generate_egg_info(path):  # type: (str) -> None
-    """Generate an egg-info in the specified base directory."""
-    # minimal PKG-INFO stub following the format defined in PEP 241
-    # required for older setuptools versions to avoid a traceback when importing pkg_resources from packages like cryptography
-    # newer setuptools versions are happy with an empty directory
-    # including a stub here means we don't need to locate the existing file or have setup.py generate it when running from source
-    pkg_info = '''
-Metadata-Version: 1.0
-Name: ansible
-Version: %s
-Platform: UNKNOWN
-Summary: Radically simple IT automation
-Author-email: info@ansible.com
-License: GPLv3+
-''' % get_ansible_version()
-
-    pkg_info_path = os.path.join(path, 'ansible_core.egg-info', 'PKG-INFO')
-
-    if os.path.exists(pkg_info_path):
-        return
-
-    write_text_file(pkg_info_path, pkg_info.lstrip(), create_directories=True)
+def generate_dist_info():  # type: () -> None
```

Comment:
@mattclay Does this change look okay to backport? I wanted to add a test and the distribution has no requirements with only the egg-info stub.

Reply:
Why is this change needed?

Reply:
The previous stub didn't fully replicate what an egg_info or dist_info would include. So uses of `importlib.metadata` were failing where we needed to query information that was not included in the stub. I recommended generating an actual output, otherwise we'd always be playing catch up to keep it properly in sync with expectations.

Reply:
Unfortunately `setup.py` isn't available once ansible is installed, so we'll need a different solution.

Reply:
For that case could we just use the dist_info already generated by setuptools/pip?

Generate it when needed, and copy when available? I don't know the full flow of `ansible-test` for this area of the code to know if that's feasible.

Reply:
We might be able to figure something out, but that doesn't seem like a change we'd want to include in a backport.

What is missing from the current stub?

Reply:
@mattclay The python requirements are missing. This PR is trying to check the resolvelib version to determine if ansible-galaxy can use it, to preemptively avoid the traceback. I could possible add exception handling instead (TypeError would be the obvious one with the changing function signatures in resolvelib), but making sure all of the unsupported versions would be covered by exception handling doesn't seem practical.

So, I either need to hardcode the supported versions of resolvelib (original attempt, would require code to be updated whenever the requirements are updated), or get the current requirements from the dist-info (current implementation, no upkeep required). 

This patch falls back to the current devel behavior (try to resolve collection dependencies regardless of resolvelib version) if the ansible-core distribution info is wrong (either incomplete with no python requirements, or missing altogether if no 'ansible-core' dist is found). With only the current stub, importlib.metadata.distribution('ansible-core').requires is None, so the ansible-galaxy-collection integration test can not use the bugfix at all.

I didn't want to, but maybe it would be fine to backport without tests?

Reply:
I think a hard-coded min and max version for resolvelib makes sense. Multiple files already need to be updated any time the supported version range changes -- adding one more doesn't seem like a big deal.

Reply:
I'd rather not add more of it just because it's done elsewhere and is convenient though. I'd like to remove the need to update the test suite but couldn't think of a good way to do so.